Add metricsfilter option (#17978)
pingcap/docs: clinic/clinic-user-guide-for-tiup.md

    - `-N/--node`: only collects data from a specified node. The format is `ip:port`.
    - `--include`: only collects specific types of data. The optional values are `system`, `monitor`, `log`, `config`, and `db_vars`. To include two or more types, you can use `,` as a separator between the types.
    - `--exclude`: does not collect specific types of data. The optional values are `system`, `monitor`, `log`, `config`, and `db_vars`. To exclude two or more types, you can use `,` as a separator between the types.

>     - `--metricsfilter`: only collects specified Prometheus metrics. You can specify metrics using a comma-separated list of metric prefixes. For example: `--metricsfilter=tidb,pd` collects metrics that start with `tidb` and metrics that start with `pd`.

Reviewer: ```suggestion
    - `--metricsfilter`: only collects specified Prometheus metrics. You can specify metrics using a comma-separated list of metric prefixes. For example, `--metricsfilter=tidb,pd` collects metrics that start with `tidb` and metrics that start with `pd`.
```

    - `-N/--node`: only collects data from a specified node. The format is `ip:port`.
    - `--include`: only collects specific types of data. The optional values are `system`, `monitor`, `log`, `config`, and `db_vars`. To include two or more types, you can use `,` as a separator between the types.
    - `--exclude`: does not collect specific types of data. The optional values are `system`, `monitor`, `log`, `config`, and `db_vars`. To exclude two or more types, you can use `,` as a separator between the types.
    - `--metricsfilter`: only collects specified Prometheus metrics. You can specify metrics using a comma-separated list of metric prefixes. For example, `--metricsfilter=tidb,pd` collects metrics that start with `tidb` and metrics that start with `pd`.
    
        > **Tip:**
        >

>         > To get available metric prefixes, you can query the TiDB monitoring API using the following command.

Reviewer: ```suggestion
        > To get available metric prefixes, you can query the TiDB monitoring API using the following command:
```

        >
        > ```bash

>         > $ curl -s 'http://${prometheus-host}:${prometheus-port}/api/v1/label/__name__/values' | jq -r '.data[]' | cut -d\_ -f1 | uniq -c | sort -rn

Reviewer: ```suggestion
        > curl -s 'http://${prometheus-host}:${prometheus-port}/api/v1/label/__name__/values' | jq -r '.data[]' | cut -d\_ -f1 | uniq -c | sort -rn
```

>         > `$ curl -s 'http://${prometheus-host}:${prometheus-port}/api/v1/label/__name__/values' | jq -r '.data[]' | cut -d\_ -f1 | uniq -c | sort -rn`

Reviewer: ```suggestion
        > $ curl -s 'http://${prometheus-host}:${prometheus-port}/api/v1/label/__name__/values' | jq -r '.data[]' | cut -d\_ -f1 | uniq -c | sort -rn
```

    - `-N/--node`: only collects data from a specified node. The format is `ip:port`.
    - `--include`: only collects specific types of data. The optional values are `system`, `monitor`, `log`, `config`, and `db_vars`. To include two or more types, you can use `,` as a separator between the types.
    - `--exclude`: does not collect specific types of data. The optional values are `system`, `monitor`, `log`, `config`, and `db_vars`. To exclude two or more types, you can use `,` as a separator between the types.

>     - `--metricsfilter`: Comma separated list of prefixes of Prometheus metrics to collect. For example: `--metricsfilter=tidb,pd` would collect metrics that start with `tidb` and metrics that start with `pd`.

Author: No as there isn't a fixed list. This is because these are prefixes and not a list of components.

With a TiUP Playground:
![image](https://github.com/user-attachments/assets/4c9d69aa-1aa8-49e6-a3be-fc2e6a449b39)

So `tidb` would match `tidb_server_query_total` but `tidb_server` would also be a valid prefix.

>     - `--metricsfilter`: Comma separated list of prefixes of Prometheus metrics to collect. For example: `--metricsfilter=tidb,pd` would collect metrics that start with `tidb` and metrics that start with `pd`.

Author: Some idea about how many metrics there are and what prefixes could be used:
```
$ curl -s 'http://127.0.0.1:9090/api/v1/label/__name__/values' | jq -r '.data[]' | cut -d\_ -f1 | uniq -c | sort -rn
    446 tikv
    365 tidb
    178 prometheus
    142 pd
    112 etcd
    107 go
     40 raft
     24 resource
      7 process
      7 grpc
      6 net
      4 scrape
      3 br
      2 promhttp
      2 os
      1 up
      1 tso
      1 service
```

>     - `--metricsfilter`: Comma separated list of prefixes of Prometheus metrics to collect. For example: `--metricsfilter=tidb,pd` would collect metrics that start with `tidb` and metrics that start with `pd`.

Reviewer: ```suggestion
    - `--metricsfilter`: only collects specified Prometheus metrics. You can specify Prometheus metrics using a comma-separated list of metric prefixes. For example: `--metricsfilter=tidb,pd` collects metrics that start with `tidb` and metrics that start with `pd`.
    
    > **Tip:**
    >
    > To get available metric prefixes, you can run `$ curl -s 'http://${prometheus-host}:${prometheus-port}/api/v1/label/__name__/values' | jq -r '.data[]' | cut -d\_ -f1 | uniq -c | sort -rn` to query the TiDB monitoring API.
```
